java-client/src/main/java/org/opensearch/client/transport/aws/AwsV2OpenSearchTransport.java

```diff
@@ -0,0 +1,427 @@
+                DefaultCredentialsProvider.builder().build();
+        this.transportOptions = options != null ? options : TransportOptions.builder().build();
+    }
+
+    /**
+     * Create an {@link OpenSearchTransport} with an ASYNCHRONOUS AWS Http client
+     * <p>
+     * Note that synchronous OpenSearch requests sent through this transport will be dispatched
+     * using the asynchronous client, but the calling thread will block until they are complete.
+     *
+     * @param asyncHttpClient HTTP client to use for OpenSearch requests
+     * @param host            The target host
+     * @param credentials     Credentials provider to user for signing requests.  Specify null to use
+     *                        the default credentials provider chain.
+     * @param mapper mapper used to serialize and deserialize requests and responses.  Specify null to
+     *               get a default mapper.
+     * @param options Options that apply to all requests.  Can be null
+     */
+    public AwsV2OpenSearchTransport(
+            @Nonnull SdkAsyncHttpClient asyncHttpClient,
+            @Nonnull String host,
+            @CheckForNull AwsCredentialsProvider credentials,
+            @CheckForNull JsonpMapper mapper,
+            @CheckForNull TransportOptions options) {
+        Objects.requireNonNull(asyncHttpClient, "AwsV2OpenSearchTransport asyncHttpClient must not be null");
+        Objects.requireNonNull(host, "AwsV2OpenSearchTransport host must not be null");
+        this.httpClient = null;
+        this.asyncHttpClient = asyncHttpClient;
+        this.host = host;
+        this.mapper = mapper != null ? mapper : new JacksonJsonpMapper();
+        this.credentials = credentials != null ? credentials :
+                DefaultCredentialsProvider.builder().build();
+        this.transportOptions = options != null ? options : TransportOptions.builder().build();
+    }
+
+    @Override
+    public <RequestT, ResponseT, ErrorT> ResponseT performRequest(
+            RequestT request,
+            Endpoint<RequestT, ResponseT, ErrorT> endpoint,
+            @Nullable TransportOptions options
+    ) throws IOException {
+
+        if (httpClient == null) {
+            try {
+                return performRequestAsync(request, endpoint, options).get();
+            } catch (ExecutionException e) {
+                Throwable cause = e.getCause();
+                if (cause != null) {
+                    if (cause instanceof IOException) {
+                        throw (IOException) cause;
+                    }
+                    if (cause instanceof RuntimeException) {
+                        throw (RuntimeException) cause;
+                    }
+                    throw new RuntimeException(cause);
+                }
+                throw new RuntimeException(e);
+            } catch (InterruptedException e) {
+                throw new RuntimeException("HttpRequest was interrupted", e);
+            }
+        }
```

Comment:
Maybe we can refactor this out into a private method that handles errors? Split the bottom into `performRequestSync`?

Reply:
This exception handling is specific to the case where an async client is used to service a synchronous request, so I'd like to leave it there.  I can split out the bottom of the two `performRequest...` methods into private ones and remove the mutual recursion, though.